libraries/ESP8266HTTPClient/examples/DigestAuthorization/DigestAuthorization.ino

```diff
@@ -0,0 +1,117 @@
+#include <ESP8266WiFi.h>
+#include <ESP8266WiFiMulti.h>
+
+#include <ESP8266HTTPClient.h>
+
+#define USE_SERIAL Serial
+
+ESP8266WiFiMulti WiFiMulti;
+
+String _exractParam(String& authReq, const String& param, const char delimit){
+	int _begin = authReq.indexOf(param);
+	if (_begin==-1) return "";
+	return authReq.substring(_begin+param.length(),authReq.indexOf(delimit,_begin+param.length()));
+}
+
+void setup() {
+
+	USE_SERIAL.begin(9600);
+	// USE_SERIAL.setDebugOutput(true);
+
+	USE_SERIAL.println();
+	USE_SERIAL.println();
+	USE_SERIAL.println();
+
+	for(uint8_t t = 4; t > 0; t--) {
+		USE_SERIAL.printf("[SETUP] WAIT %d...\n", t);
+		USE_SERIAL.flush();
+		delay(1000);
+	}
+
+	WiFi.mode(WIFI_STA);
+	WiFiMulti.addAP("Niligo-Prism-yyr9uma", "");
+
+}
+
+void loop() {
+	// wait for WiFi connection
+	if((WiFiMulti.run() == WL_CONNECTED)) {
+
+		HTTPClient http;
+
+		USE_SERIAL.print("[HTTP] begin...\n");
+		// configure traged server and url
+
+
+		// http.begin("http://admin:admin@192.168.100.1/api/state?power=1");
+		http.begin("http://192.168.100.1/api/state?power=0");
+		/*
+		// or
+		// or
+		http.begin("http://192.168.1.12/test.html");
+		http.setAuthorization("dXNlcjpwYXN3b3Jk");
+		 */
+
+
+		const char *keys[] = {"WWW-Authenticate", "Server", "Connection"};
+		http.collectHeaders(keys, 3);
+
+		USE_SERIAL.print("[HTTP] GET...\n");
+		// start connection and send HTTP header
+		int httpCode = http.GET();
+
+		if (httpCode > 0) {
+			String authReq = http.header("WWW-Authenticate");
+			USE_SERIAL.println(authReq);
+
+			// file found at server
+			String payload = http.getString();
+			USE_SERIAL.println(payload);
+
+			String username = "admin";
+			String password = "admin";
+			
+			// extracting required parameters for RFC 2069 simpler Digest
+			// String _realm    = _exractParam(authReq, "realm=\"", '"');
+			String _realm = "Protected";
```

Comment:
Can you explain why it says "extracting required parameters" and then the `extractParam` call is commented out?